x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/index/store/cache/CacheFileTests.java

```diff
@@ -124,9 +130,40 @@ public void testDeleteOnCloseAfterLastRelease() throws Exception {
         assertFalse(Files.exists(file));
     }
 
-    class TestEvictionListener implements EvictionListener {
+    public void testConcurrentAccess() throws Exception {
+        final Path file = createTempDir().resolve("file.cache");
+        final CacheFile cacheFile = new CacheFile("test", randomLongBetween(1, 100), file);
+
+        final TestEvictionListener evictionListener = new TestEvictionListener();
+        assertTrue(cacheFile.acquire(evictionListener));
+        final long length = cacheFile.getLength();
+        final DeterministicTaskQueue deterministicTaskQueue = new DeterministicTaskQueue(
+            builder().put(NODE_NAME_SETTING.getKey(), getTestName()).build(),
+            random()
+        );
+        final ThreadPool threadPool = deterministicTaskQueue.getThreadPool();
+        final CompletableFuture<Integer> readFuture = cacheFile.populateAndRead(
```

Comment:
Can we also (sometimes, randomly) call readIfAvailableOrPending?

Reply:
sure thing adding it :)